addons/stock/stock.py

```diff
@@ -895,8 +895,10 @@ def test_assigned(self, cr, uid, ids):
                 if all([x.state != 'waiting' for x in pick.move_lines]):
                     return True
             for move in pick.move_lines:
-                if (move.state) == 'waiting':
-                    move.check_assign()
```

Comment:
Can't you simply past here the context?: `move.check_assign(context={'picking_trg_write': False})`